DateFormat, TimeFormat, TimestampFormat support

snuyanzin opened the pull request:
The PR adds possibility to specify date/time/timestamp format. As I understand there is already existing issue for that #66

julianhyde commented:
Thanks for this! A couple of comments:
* In `sqlline.properties` can you move `timeformat`, timestampformat` and `dateformat so that they are in alphabetical order, like other properties;
* For each property, can you give a simple example (e.g. 'Y-M-D') and also mention that you can assign "default" to set it back to the default.
* Can you please add a test. Something like `SqlLineArgsTest.testTablesCsv`. You should be able to call `!set dateformat ...` in the script.

An enhancement request. (I will accept the PR without it, but please do it if it's not too hard.) Can you validate the format strings when they are set? Currently they get validated when you run the query, which is confusing:

```
> !connect jdbc:hsqldb:res:scott SCOTT TIGER
> !set dateformat DD-MON-YYYY
> select * from emp;
java.lang.IllegalArgumentException: Illegal pattern character 'N'
	at java.text.SimpleDateFormat.compile(SimpleDateFormat.java:826)
	at java.text.SimpleDateFormat.initialize(SimpleDateFormat.java:634)
	at java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:605)
	at java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:580)
```

snuyanzin commented:
Thank you for your response and comments
- Done sorting of new properties in `sqlline.propeprties`
- Specified one example for each new property + added a link to `SimpleDateFormat` javadoc with detail description of possible patterns.
- Added two types of tests: failed and success
- About enhancement request: the only relatively cheap solution I could think is trying to format whatever date/time with new mask and failing in case of wrong pattern and previous valid pattern will be used instead. I did such implementation in this PR however to have more friendly exception message I changed `sqlline.SqlLineOpts#set` to return cause if it exists. After that the behavior will be like that 
```
> !set dateformat DD-MON-YYYY
Error setting configuration: dateformat: java.lang.IllegalArgumentException: Illegal pattern character 'O'
```

julianhyde commented:
Merged as a75929dcb9a9f7af6f891d8c737c8c2083e545d5, and marked #66 fixed. Many thanks, @snuyanzin!
